src/components/KyberAITokenBanner/index.tsx

```diff
@@ -0,0 +1,252 @@
+import Column from 'components/Column'
+import ApeIcon from 'components/Icons/ApeIcon'
+import Row, { RowBetween, RowFit } from 'components/Row'
+import { APP_PATHS } from 'constants/index'
+import { NativeCurrencies } from 'constants/tokens'
+import { useActiveWeb3React } from 'hooks'
+import useMixpanel, { MIXPANEL_TYPE } from 'hooks/useMixpanel'
+import useTheme from 'hooks/useTheme'
+import KyberScoreMeter from 'pages/TrueSightV2/components/KyberScoreMeter'
+import { NETWORK_TO_CHAINID } from 'pages/TrueSightV2/constants'
+import { SUPPORTED_NETWORK_KYBERAI } from 'pages/TrueSightV2/constants/index'
+import { useTokenDetailQuery } from 'pages/TrueSightV2/hooks/useKyberAIData'
+import { ITokenOverview } from 'pages/TrueSightV2/types'
+import { calculateValueToColor } from 'pages/TrueSightV2/utils'
+import { useIsWhiteListKyberAI } from 'state/user/hooks'
+import { MEDIA_WIDTHS } from 'theme'
+
+const KyberAITokenBanner = ({
+  currencyIn,
+  currencyOut,
+}: {
+  currencyIn: Currency | undefined
+  currencyOut: Currency | undefined
+}) => {
+  const { chainId, account } = useActiveWeb3React()
+  const { isWhiteList } = useIsWhiteListKyberAI()
+  const navigate = useNavigate()
+  const { mixpanelHandler } = useMixpanel()
+  const chain = Object.keys(NETWORK_TO_CHAINID).find(i => NETWORK_TO_CHAINID[i] === chainId)
+  const above768 = useMedia(`(min-width:${MEDIA_WIDTHS.upToSmall}px)`)
+  const theme = useTheme()
+
+  const token0 = currencyIn?.wrapped
+  const token1 = currencyOut?.wrapped
+
+  const { data: tokenInputOverview, isFetching: fetching0 } = useTokenDetailQuery(
+    { address: token0?.address, chain },
+    { skip: !token0?.address || !account || !isWhiteList, refetchOnMountOrArgChange: true },
+  )
+
+  const { data: tokenOutputOverview, isFetching: fetching1 } = useTokenDetailQuery(
+    { address: token1?.address, chain },
+    { skip: !token1?.address || !account || !isWhiteList, refetchOnMountOrArgChange: true },
+  )
+
+  const { data: tokenNativeOverview, isFetching: fetching2 } = useTokenDetailQuery(
+    { address: NativeCurrencies[chainId].wrapped.address, chain },
+    { skip: !account || !isWhiteList, refetchOnMountOrArgChange: true },
+  )
+
+  const isFetching = fetching0 || fetching1 || fetching2
+
+  const token: ITokenOverview | undefined = useMemo(() => {
+    if (isFetching) return undefined
+
+    const token =
+      tokenInputOverview?.kyberScore && tokenInputOverview.kyberScore.label !== ''
+        ? tokenInputOverview
+        : tokenOutputOverview?.kyberScore && tokenOutputOverview.kyberScore.label !== ''
+        ? tokenOutputOverview
+        : tokenNativeOverview?.kyberScore && tokenNativeOverview.kyberScore.label !== ''
+        ? tokenNativeOverview
+        : undefined
+
+    return token
+  }, [isFetching, tokenInputOverview, tokenOutputOverview, tokenNativeOverview])
+
+  if (!token || !isWhiteList || !account) return null

```

Comment:
Can we add a skeleton loading for this. currently, UI is jerky when I change the token

Reply:
will improve this later